lib/web/terminal.go

```diff
@@ -310,6 +303,16 @@ func (t *TerminalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	sessionMetadataResponse, err := json.Marshal(siteSessionGenerateResponse{Session: t.sessionData})
+
+	// If the displayLogin is set then use it in the session metadata instead of the
+	// login name used in the SSH connection. This is specifically for the use case
+	// when joining a session to avoid displaying "-teleport-internal-join" as the username.
+	if t.displayLogin != "" {
+		sessionDataTemp := t.sessionData
+		sessionDataTemp.Login = t.displayLogin
+		sessionMetadataResponse, err = json.Marshal(siteSessionGenerateResponse{Session: sessionDataTemp})
```

Comment:
You're overwriting the `err` from line 305 here without first checking if there was an error before - is that what you intended?

Reply:
This was intentional. Both lines serve the same purpose of marshaling the response into json, and so even in the event that `305` errors out but `313` passes, we will still end up with a valid marshaled `sessionMetadaResponse` and so the error on `317` would be incorrect. I also can't imagine a scenario where `305` would fail and `313` wouldn't since they're doing the same thing with just one string field altered in the latter.

Reply:
Since once is overwriting the other anyway, why not make it:

```
if t.displayLogin != "" {
	sessionDataTemp := t.sessionData
	sessionDataTemp.Login = t.displayLogin
	sessionMetadataResponse, err = json.Marshal(siteSessionGenerateResponse{Session: sessionDataTemp})
} else {
	sessionMetadataResponse, err := json.Marshal(siteSessionGenerateResponse{Session: t.sessionData})
}
```

This way we save ourselves an extra `json.Marshal` call when `displayLogin` is set.